src/javascript/app/common/attach_dom/popup.js

```diff
@@ -25,7 +26,19 @@ const callback = (options) => {
     const div      = createElement('div', { html: cache[options.url] });
     const lightbox = createElement('div', { id: options.popup_id, class: 'lightbox' });
     lightbox.appendChild(div.querySelector(options.content_id));
-    document.body.appendChild(lightbox);
+    lightbox.addEventListener('DOMNodeRemoved', (e) => {
+        if (!popup_queue.length || e.target.className !== 'lightbox') return;
+        document.body.appendChild(popup_queue.pop()); // show popup in queue
+    });
+
+    const has_lightbox = document.querySelector('.lightbox');
+    if (has_lightbox) {
+        // store this popup in queue to show it after the previous popup has been removed
+        // to avoid having multiple popup showing at the same time
+        popup_queue.push(lightbox);
+    } else {
+        document.body.appendChild(lightbox);
+    }
```

Comment:
I hate having to create a queue of popups, but nice workaround.

Reply:
we should show the third party redirect first, then the telegram installation popup. as of now it shows the telegram popup, then opens the new tab, then when you go back to binary tab you see the third party redirect popup.

Reply:
not sure if it's better to combine them rather than show two separate ones. can you comment as well @ashkanx

Reply:
yeah, i think it's makes more sense to show third party redirect first then the telegram installation popup too. better to combine them in this case